src/util/projection.ts

```diff
@@ -0,0 +1,51 @@
+import {ProjectionDefinitionT, ProjectionDefinitionSpecification, PropertyValueSpecification} from '../types.g';
+
+export default class ProjectionDefinition extends Array {
```

Comment:
I'm not sure it makes sense to inherit from `Array`... methods like include/shift/pop are not relevant for this class...